File: data/multilang_comparison/program_cpp_array/src/func4.cpp
#include "program_cpp_array.hpp" 
Array* func4(Array_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   Array* array110;
   if (pCounter > 0) {
      array110 = vars->data[--pCounter];
      array110->refC++;
      DEBUG_COPY(array110->id);
   } else {
      array110 = new Array();
      array110->size = 209;
      array110->refC = 1;
      array110->id = 110;
      array110->data = new unsigned int[array110->size]();
      DEBUG_NEW(array110->id);
   }
   unsigned int loop33 = 0;
   unsigned int loopLimit33 = (50)/3 + 1;
   for(; loop33 < loopLimit33; loop33++) {
      if(PATH0 & 1) {
         Array* array111;
         if (pCounter > 0) {
            array111 = vars->data[--pCounter];
            array111->refC++;
            DEBUG_COPY(array111->id);
         } else {
            array111 = new Array();
            array111->size = 581;
            array111->refC = 1;
            array111->id = 111;
            array111->data = new unsigned int[array111->size]();
            DEBUG_NEW(array111->id);
         }
         unsigned int loop34 = 0;
         unsigned int loopLimit34 = (50)/4 + 1;
         for(; loop34 < loopLimit34; loop34++) {
            for (int i = 0; i < array111->size; i++) {
                array111->data[i]--;
            }
            for (int i = 0; i < array110->size; i++) {
                if (array110->data[i] == 32) {
                    return array110;
                }
            }
         }
         Array_param params0;
         params0.size = 2;
         Array* data_params0[2];
         params0.data = data_params0;
         params0.data[0] = array110;
         params0.data[1] = array111;
         Array* array112 = func24(&params0, get_path(), loopsFactor);
         DEBUG_RETURN(array112->id);
         if(params0.size > 0) {
         	 params0.data.clear();
         }
         array112->refC--;
         if(array112->refC == 0) {
            delete[] array112->data;
            array112->data = nullptr;
            DEBUG_FREE(array112->id);
            delete array112;
         }
         array111->refC--;
         if(array111->refC == 0) {
            delete[] array111->data;
            array111->data = nullptr;
            DEBUG_FREE(array111->id);
            delete array111;
         }
      }
      else {
         Array_param params0;
         params0.size = 1;
         Array* data_params0[1];
         params0.data = data_params0;
         params0.data[0] = array110;
         Array* array113 = func16(&params0, get_path(), loopsFactor);
         DEBUG_RETURN(array113->id);
         if(params0.size > 0) {
         	 params0.data.clear();
         }
         array113->refC--;
         if(array113->refC == 0) {
            delete[] array113->data;
            array113->data = nullptr;
            DEBUG_FREE(array113->id);
            delete array113;
         }
      }
      Array* array114;
      if (pCounter > 0) {
         array114 = vars->data[--pCounter];
         array114->refC++;
         DEBUG_COPY(array114->id);
      } else {
         array114 = new Array();
         array114->size = 769;
         array114->refC = 1;
         array114->id = 114;
         array114->data = new unsigned int[array114->size]();
         DEBUG_NEW(array114->id);
      }
      Array_param params0;
      params0.size = 2;
      Array* data_params0[2];
      params0.data = data_params0;
      params0.data[0] = array110;
      params0.data[1] = array114;
      Array* array115 = func17(&params0, loopsFactor);
      DEBUG_RETURN(array115->id);
      if(params0.size > 0) {
      	 params0.data.clear();
      }
      if(PATH0 & 2) {
         Array* array116;
         if (pCounter > 0) {
            array116 = vars->data[--pCounter];
            array116->refC++;
            DEBUG_COPY(array116->id);
         } else {
            array116 = new Array();
            array116->size = 850;
            array116->refC = 1;
            array116->id = 116;
            array116->data = new unsigned int[array116->size]();
            DEBUG_NEW(array116->id);
         }
         unsigned int loop35 = 0;
         unsigned int loopLimit35 = (50)/4 + 1;
         for(; loop35 < loopLimit35; loop35++) {
            for (int i = 0; i < array116->size; i++) {
                array116->data[i]--;
            }
            for (int i = 0; i < array110->size; i++) {
                if (array110->data[i] == 42) {
                    return array110;
                }
            }
         }
         Array_param params1;
         params1.size = 4;
         Array* data_params1[4];
         params1.data = data_params1;
         params1.data[0] = array110;
         params1.data[1] = array114;
         params1.data[2] = array115;
         params1.data[3] = array116;
         Array* array117 = func28(&params1, get_path(), loopsFactor);
         DEBUG_RETURN(array117->id);
         if(params1.size > 0) {
         	 params1.data.clear();
         }
         array117->refC--;
         if(array117->refC == 0) {
            delete[] array117->data;
            array117->data = nullptr;
            DEBUG_FREE(array117->id);
            delete array117;
         }
         array116->refC--;
         if(array116->refC == 0) {
            delete[] array116->data;
            array116->data = nullptr;
            DEBUG_FREE(array116->id);
            delete array116;
         }
      }
      else {
         Array_param params1;
         params1.size = 3;
         Array* data_params1[3];
         params1.data = data_params1;
         params1.data[0] = array110;
         params1.data[1] = array114;
         params1.data[2] = array115;
         Array* array118 = func22(&params1, get_path(), loopsFactor);
         DEBUG_RETURN(array118->id);
         if(params1.size > 0) {
         	 params1.data.clear();
         }
         array118->refC--;
         if(array118->refC == 0) {
            delete[] array118->data;
            array118->data = nullptr;
            DEBUG_FREE(array118->id);
            delete array118;
         }
      }
      for (int i = 0; i < array115->size; i++) {
          if (array115->data[i] == 67) {
              return array115;
          }
      }
      array115->refC--;
      if(array115->refC == 0) {
         delete[] array115->data;
         array115->data = nullptr;
         DEBUG_FREE(array115->id);
         delete array115;
      }
      array114->refC--;
      if(array114->refC == 0) {
         delete[] array114->data;
         array114->data = nullptr;
         DEBUG_FREE(array114->id);
         delete array114;
      }
   }
   for (int i = 0; i < array110->size; i++) {
       array110->data[i]--;
   }
   for (int i = 0; i < array110->size; i++) {
       if (array110->data[i] == 61) {
           return array110;
       }
   }
   return array110;
}

File: data/multilang_comparison/program_julia/src/func4.jl
include("program_julia_head.jl")
function func4(vars::JArray_param, PATH0::UInt64, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   array110 = nothing
   if pCounter > 1
       array110 = vars.data[pCounter]
       pCounter -= 1
       array110.refC += 1
   else
       array110 = JArray(zeros(Int, 209), 209, 1, 110)
   end
   loopLimit33 = (50)/3 + 1
   for i in 1:loopLimit33 
      if (PATH0 & 1) != 0
         array111 = nothing
         if pCounter > 1
             array111 = vars.data[pCounter]
             pCounter -= 1
             array111.refC += 1
         else
             array111 = JArray(zeros(Int, 581), 581, 1, 111)
         end
         loopLimit34 = (50)/4 + 1
         for i in 1:loopLimit34 
            for i in 0:(array111.size-1) 
               array111.data[i+1] -= 1
            end
            for i in 1:array110.size
                if array110.data[i] == 32
                        return array110
               end
            end
         end
         params0 = JArray_param{Int}()
         params0.size = 2
         push!(params0.data, array110)
         params0.size += 1
         push!(params0.data, array111)
         params0.size += 1
         array112 = func24(params0, get_path(), loopsFactor);
         array112.refC -= 1
         if array112.refC == 0
             empty!(array112.data)
             array112 = nothing
         end
         array111.refC -= 1
         if array111.refC == 0
             empty!(array111.data)
             array111 = nothing
         end
      
      else 
         params0 = JArray_param{Int}()
         params0.size = 1
         push!(params0.data, array110)
         params0.size += 1
         array113 = func16(params0, get_path(), loopsFactor);
         array113.refC -= 1
         if array113.refC == 0
             empty!(array113.data)
             array113 = nothing
         end
      end
      array114 = nothing
      if pCounter > 1
          array114 = vars.data[pCounter]
          pCounter -= 1
          array114.refC += 1
      else
          array114 = JArray(zeros(Int, 769), 769, 1, 114)
      end
      params0 = JArray_param{Int}()
      params0.size = 2
      push!(params0.data, array110)
      params0.size += 1
      push!(params0.data, array114)
      params0.size += 1
      array115 = func17(params0, loopsFactor);
      if (PATH0 & 2) != 0
         array116 = nothing
         if pCounter > 1
             array116 = vars.data[pCounter]
             pCounter -= 1
             array116.refC += 1
         else
             array116 = JArray(zeros(Int, 850), 850, 1, 116)
         end
         loopLimit35 = (50)/4 + 1
         for i in 1:loopLimit35 
            for i in 0:(array116.size-1) 
               array116.data[i+1] -= 1
            end
            for i in 1:array110.size
                if array110.data[i] == 42
                        return array110
               end
            end
         end
         params1 = JArray_param{Int}()
         params1.size = 4
         push!(params1.data, array110)
         params1.size += 1
         push!(params1.data, array114)
         params1.size += 1
         push!(params1.data, array115)
         params1.size += 1
         push!(params1.data, array116)
         params1.size += 1
         array117 = func28(params1, get_path(), loopsFactor);
         array117.refC -= 1
         if array117.refC == 0
             empty!(array117.data)
             array117 = nothing
         end
         array116.refC -= 1
         if array116.refC == 0
             empty!(array116.data)
             array116 = nothing
         end
      
      else 
         params1 = JArray_param{Int}()
         params1.size = 3
         push!(params1.data, array110)
         params1.size += 1
         push!(params1.data, array114)
         params1.size += 1
         push!(params1.data, array115)
         params1.size += 1
         array118 = func22(params1, get_path(), loopsFactor);
         array118.refC -= 1
         if array118.refC == 0
             empty!(array118.data)
             array118 = nothing
         end
      end
      for i in 1:array115.size
          if array115.data[i] == 67
                  return array115
         end
      end
      array115.refC -= 1
      if array115.refC == 0
          empty!(array115.data)
          array115 = nothing
      end
      array114.refC -= 1
      if array114.refC == 0
          empty!(array114.data)
          array114 = nothing
      end
   end
   for i in 0:(array110.size-1) 
      array110.data[i+1] -= 1
   end
   for i in 1:array110.size
       if array110.data[i] == 61
               return array110
      end
   end
   return array110;
end

